js-packages/profiler-lib/src/profiler.ts

```diff
@@ -9,10 +9,17 @@ import { MetadataSelector } from './metadataSelection.js';
 export interface ProfilerConfig {
     /** Container element for the graph visualization */
     graphContainer: HTMLElement;
-    /** Container element for the selector UI controls */
-    selectorContainer: HTMLElement;
     /** Container element for the navigator minimap */
     navigatorContainer: HTMLElement;
+
+    // UI element references (dependency injection)
+    /** Metric selector dropdown element */
+    metricSelector: HTMLSelectElement;
```

Comment:
These should be at the very least bundled in a single higher-level object SelectorConfig.
But I really find it very weird to spread ui state from many components into a separate config class.
Moreover, Config is not the right name, this should have some kind of UI in the name, since it's not configuration, it is a collection of containers for UI elements.